projects/ngx-d3timeline/src/lib/options/selectors/options.selectors.ts

```diff
@@ -20,20 +15,30 @@ export const selectResourceOrientation = createSelector(
   flipOrientation
 );
 
-export const selectAcivityPadding = createSelector(
-  selectTypeActivityPadding,
-  selectActivityOptionsPadding,
-  (typeActivityPadding, activityOptionsPadding) =>
-    getActivityPadding.bind(null, typeActivityPadding, activityOptionsPadding)
-);
+const selectActivityOption = <T>(
+  key: string
+): MemoizedSelector<(type: string) => T> =>
+  createSelector(
+    selectTypeActivityOption<T>(key),
+    selectGlobalActivityOption<T>(key),
+    (typeActivityOption, globalActivityOption) =>
+      getActivityOption.bind(
+        null,
+        typeActivityOption,
+        globalActivityOption
+      ) as (type: string) => T
+  );
+
+export const selectAcivityPadding = selectActivityOption<number>('padding');
 
-export const selectActivityStrokeWidth = createSelector(
-  selectTypeActivityStrokeWidth,
-  selectActivityOptionsStrokeWidth,
-  (typeActivityStrokeWidth, activityOptionsStrokeWidth) =>
-    getActivityPadding.bind(
-      null,
-      typeActivityStrokeWidth,
-      activityOptionsStrokeWidth
-    )
+export const selectActivityStrokeWidth = selectActivityOption<number>(
+  'strokeWidth'
 );
+
+function getActivityOption<T>(
```

Comment:
Again pretty much what I had but if there's anywhere else that we use this options precedence (axis?) this can be made more generic to handle all cases not just activity options.

```ts
export function getRelevantOption<TOption>(
  specialisedOptionFunc: (type: string) => TOption,
  globalOptionValue: TOption,
  type: string
) {
  return specialisedOptionFunc(type) || globalOptionValue;
}
```